ingest_api/src/main.py

```diff
@@ -0,0 +1,219 @@
+    tags=["Ingestion"],
+)
+def get_ingestion(
+    ingestion: schemas.Ingestion = Depends(dependencies.fetch_ingestion),
+) -> schemas.Ingestion:
+    """
+    Gets the status of an ingestion.
+    """
+    return ingestion
+
+
+@app.patch(
+    "/ingestions/{ingestion_id}",
+    response_model=schemas.Ingestion,
+    tags=["Ingestion"],
+)
+def update_ingestion(
+    update: schemas.UpdateIngestionRequest,
+    ingestion: schemas.Ingestion = Depends(dependencies.fetch_ingestion),
+    db: services.Database = Depends(dependencies.get_db),
+):
+    """
+    Updates the STAC item with the provided item.
+    """
+    updated_item = ingestion.copy(update=update.dict(exclude_unset=True))
+    return updated_item.save(db)
+
+
+@app.delete(
+    "/ingestions/{ingestion_id}",
+    response_model=schemas.Ingestion,
+    tags=["Ingestion"],
+)
+def cancel_ingestion(
+    ingestion: schemas.Ingestion = Depends(dependencies.fetch_ingestion),
+    db: services.Database = Depends(dependencies.get_db),
+) -> schemas.Ingestion:
+    """
+    Cancels an ingestion in queued state."""
+    if ingestion.status != schemas.Status.queued:
+        raise HTTPException(
+            status_code=400,
+            detail=(
+                "Unable to delete ingestion if status is not "
+                f"{schemas.Status.queued}"
+            ),
+        )
+    return ingestion.cancel(db)
+
+
+@app.post(
+    "/collections",
+    tags=["Collection"],
+    status_code=201,
+    dependencies=[Depends(auth.get_username)],
+)
+def publish_collection(collection: schemas.DashboardCollection):
+    """
+    Publish a collection to the STAC database.
+    """
+    # pgstac create collection
+    try:
+        collection_publisher.ingest(collection)
+        return {f"Successfully published: {collection.id}"}
+    except Exception as e:
+        raise HTTPException(
+            status_code=400,
+            detail=(f"Unable to publish collection: {e}"),
+        )
+
+
+@app.delete(
+    "/collections/{collection_id}",
+    tags=["Collection"],
+    dependencies=[Depends(auth.get_username)],
+)
+def delete_collection(collection_id: str):
+    """
+    Delete a collection from the STAC database.
+    """
+    try:
+        collection_publisher.delete(collection_id=collection_id)
+        return {f"Successfully deleted: {collection_id}"}
+    except Exception as e:
+        print(e)
+        raise HTTPException(status_code=400, detail=(f"{e}"))
+
+
+@app.post(
+    "/items",
+    tags=["Items"],
+    status_code=201,
+    dependencies=[Depends(auth.get_username)],
+)
+def publish_item(item: schemas.Item):
+    """
+    Publish a collection to the STAC database.
+    """
+    # pgstac create collection
+    try:
+        collection_publisher.ingest(item)
+        return {f"Successfully published: {item.id}"}
+    except Exception as e:
+        raise HTTPException(
+            status_code=400,
+            detail=(f"Unable to publish collection: {e}"),
+        )
```

Comment:
I'm confused, what is this for?

Reply:
I messed up a refactor here, but this is meant to be an endpoint to bypass the normal ingestion process and load any valid STAC item into pgstac.